File: Advanced Programming in C++/cashregister.cpp
#include<bits/stdc++.h>
using namespace std;
int main() {
    float item1;
    float item2;
    char c;
    float tax;
    // define paraments
     
    cout << "Enter price of the first item: ";
    cin >> item1;
    cout << "Enter price of the second item: ";
    cin >> item2;
    cout << "Does customer have a club card? (Y/N): ";
    cin >> c;
    cout << "Enter tax rate, e.g. 5.5 for 5.5% tax: ";
    cin >> tax;
    // input paraments

    float base = item1 + item2;
    float discount = item1 < item2 ? item1 / 2 + item2 : item2 / 2 + item1;
    if (c == 'Y' || c == 'y') {
        discount *= 0.9;
    }
    float total = discount + discount * tax / 100;
    // caculate cost

    printf("Base price = %.2f\n", base);
    printf("Price after discounts = %.2f\n", discount);
    printf("Total price = %.2f\n", total);
    // print ans

    return 0;
}

File: Basics of Computing and Programming/cashregister.py
def main():
    price = []
    for i in range(2):
        times = "first" if not(i) else "second"
        price.append(float(input(f"Enter price of the {times} item: ")))
    price.sort()
    base_price = sum(price)
    price[0] /= 2
    discount_price = sum(price)
    if input("Does customer have a club card? (Y/N): ").lower() == 'y':
        discount_price *= 0.9
    tax_rate = float(input("Enter tax rate, e.g. 5.5 for 5.5% tax: ")) / 100
    print(f"Base price = {round(base_price, 2):.2f}")
    if discount_price != base_price:
        print(f"Price after discounts = {round(discount_price, 2):.2f}")
    print(f"Total price = {round(discount_price * (1 + tax_rate), 2):.2f}")

if __name__ == "__main__":
    main()